samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,268 @@
+It is used by client to determine which version of Wi-Fi provisioning service is running on this server.
+
+Operation Control Point
+-----------------------
+
+This characteristic is used to send Request message and get Response message.
+The client shall send an encoded message in type of Request defined in request.proto, and the server shall send an encoded message in type of Response defined in response.proto.
+
+Data Out
+--------
+
+This characteristic is used to send notifications to the client in format of Result message.
+
+Advertisement
+-------------
+
+After power up, the device will first check if it is provisioned or not.
+If so, it will load saved credentials and try to connect the AP specified in the credential.
+After 30 seconds (timeout will occur if the attempt fails), it will start BLE advertising.
+If it is not provisioned, it will advertise immediately.
+
+The advertising data contains several fields to facilitate possible provisioning attempt.
+It will contain the UUID of Wi-Fi provisioning service, and a 4-byte service data.
+
++----------+----------+-----------+----------+
+|  Byte 1  |  Byte 2  |   Byte 3  |  Byte 4  |
++==========+==========+===========+==========+
+| Version  |         Flag         |   RSSI   |
++----------+----------------------+----------+
+
+Version: 8-bit unsigned integer. The value is the version of Wi-Fi provisioning service.
+
+Flag: 16-bit little endian field. Byte 2 (first byte on air) is LSB, and Byte 3 (second byte on air) is MSB.
+
+- Bit 0: provisioning status. 1 if device is provisioned, 0 otherwise.
+
+- Bit 1: connection status. 1 if Wi-Fi is connected, 0 otherwise.
+
+- Bit 2 - 15: RFU
+
+RSSI: 8-bit signed integer. If Wi-Fi is connected, it is the signal strength. Otherwise, it is meaningless.
+
+Depending on the provisioning status, the advertising interval differs.
+If it is not provisioned, the interval is 100ms. If it is provisioned, the interval is 1 second.
+
+Configuration management
+========================
+
+Once the device receives Wi-Fi configuration, it should keep the data in proper location, and save or delete the record according to incoming command.
+Therefore, the configuration management component is to encapsulate these procedures and expose some APIs.
+It provides two functions.
+First, it manages the stored configuration, and make sure the data in RAM and data in flash are in sync.
+Second, it provides API to add and remove configuration safely in a multi-thread environment.
+
+When used in Wi-Fi provisioning task, the main thread will initialize it and try to get saved configuration.
+It is applied if there is any.
+Then, the handler may read, write, and remove configuration based on the command.
+And when certain configuration makes Wi-Fi connection successful, the API that commit this configuration is called to mark it as valid.
+
+This component saves data in non-volatile memory (NVM), which is the internal flash in nRF5x chips.
```

Comment:
We should have defined NVM far above so we can either use the abbrevation here, or keep the full phrase everywhere. I prefer just spelling it out as nonvolatile memory.

'related kconfig options are enabled': would be good to know which, and whether they are enabled by default or configurable. Good place for a link.

I think the bit about nRF5 devices (I wouldn't say chips, but am not sure) should be a note because it's not relevant to all users.

Use 'you' instead of 'the user'.

Take out 'when' from 'when during'

Reply:
The kconfig option is removed at this moment. So I will removed related sentences.